src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCaptureCollection.cs

```diff
@@ -116,6 +125,120 @@ internal Capture GetCapture(int i)
 
             return _captures[i];
         }
+
+        int IList<Capture>.IndexOf(Capture item)
+        {
+            var comparer = EqualityComparer<Capture>.Default;
+            for (int i = 0; i < Count; i++)
+            {
+                if (comparer.Equals(this[i], item))
+                    return i;
+            }
+            return -1;
+        }
+
+        void IList<Capture>.Insert(int index, Capture item)
+        {
+            throw new NotSupportedException();
+        }
```

Comment:
I think this component should follow the conventions used by the larger framework, i.e. include exception message. Both for consistency, and so we can explain in the message how to test for whether these operations are supported or not (point people to the IsReadOnly property).